mappings/net/minecraft/world/gen/feature/AbstractTreeFeature.mapping

```diff
@@ -8,7 +8,48 @@ CLASS bto net/minecraft/world/gen/feature/AbstractTreeFeature
 		ARG 3 random
 		ARG 4 pos
 		ARG 5 config
+	METHOD a setToDirt (Lbbv;Let;)V
+		ARG 1 world
+		ARG 2 pos
+	METHOD a canTreeReplace (Lbbw;Let;)Z
+		ARG 0 world
+		ARG 1 pos
 	METHOD a setBlockState (Lbbz;Let;Lbpm;)V
 		ARG 1 world
 		ARG 2 pos
 		ARG 3 state
+	METHOD a generate (Ljava/util/Set;Lbbv;Ljava/util/Random;Let;)Z
+		ARG 1 logPositions
+		ARG 2 world
+		ARG 3 random
+		ARG 4 pos
+	METHOD a setBlockState (Ljava/util/Set;Lbbz;Let;Lbpm;)V
+		ARG 1 logPositions
+		ARG 2 world
+		ARG 3 pos
+		ARG 4 state
+	METHOD b isAir (Lbbw;Let;)Z
+		ARG 0 world
+		ARG 1 pos
+	METHOD b setBlockStateWithoutUpdatingNeighbors (Lbbz;Let;Lbpm;)V
+	METHOD c isNaturalDirt (Lbbw;Let;)Z
+		ARG 0 world
+		ARG 1 pos
+	METHOD e isWater (Lbbw;Let;)Z
+		ARG 0 world
+		ARG 1 pos
+	METHOD f isLeaves (Lbbw;Let;)Z
+		ARG 0 world
+		ARG 1 pos
+	METHOD g isAirOrLeaves (Lbbw;Let;)Z
```

Comment:
Are all of these methods used generally for checking types, or maybe they're more of a "canBeX" kind of deal?

Reply:
All these static methods just do something like `return world.test(pos, blockState -> blockState.isAir() || blockState.matches(BlockTags.LEAVES))`.

Reply:
Fair. However, keep in mind it's not about what they *do*, it's about what they're *used for*, often.